paymentsheet/src/androidTest/java/com/stripe/android/paymentsheet/utils/FlowControllerTestRunner.kt

```diff
@@ -100,3 +103,42 @@ private fun runFlowControllerTest(
     assertThat(countDownLatch.await(5, TimeUnit.SECONDS)).isTrue()
     scenario.close()
 }
+
+internal class SynchronizedTestFlowController(
+    private val flowController: PaymentSheet.FlowController
+) : PaymentSheet.FlowController by flowController, IdlingResource {
+    private var isIdleNow: Boolean = false
+    private var onIdleTransitionCallback: IdlingResource.ResourceCallback? = null
+
+    override var shippingDetails: AddressDetails?
```

Comment:
We should be able to get rid of this, since we're delegating `PaymentSheet.FlowController by flowController`